--- linux-b82bf0c/arch/arm/mm/Makefile	2012-12-16 23:21:55.000000000 +0100
+++ linux-b82bf0c/arch/arm/mm/Makefile	2012-12-16 23:25:27.181375418 +0100
@@ -1,6 +1,7 @@
 #
 # Makefile for the linux arm-specific parts of the memory manager.
 #
+plus_sec := $(call as-instr,.arch_extension sec,+sec)
 
 obj-y				:= dma-mapping.o extable.o fault.o init.o \
 				   iomap.o
@@ -94,7 +95,7 @@ obj-$(CONFIG_CPU_V6K)		+= proc-v6.o
 obj-$(CONFIG_CPU_V7)		+= proc-v7.o
 
 AFLAGS_proc-v6.o	:=-Wa,-march=armv6
-AFLAGS_proc-v7.o	:=-Wa,-march=armv7-a
+AFLAGS_proc-v7.o	:=-Wa,-march=armv7-a$(plus_sec)
 
 obj-$(CONFIG_CACHE_FEROCEON_L2)	+= cache-feroceon-l2.o
 obj-$(CONFIG_CACHE_L2X0)	+= cache-l2x0.o
